### ISSUE TITLE
iOS Class Chain locator strategu

### ISSUE THREAD

**Issue by titusfortner** [watir/tap_watir#2]
XPath is inherently slow for iOS, so Watir's general approach of converting a locator hash into an XPath to locate elements in a DOM won't be performant. We will need to create an iOS specific locator class for Class Chains (and/or Predicate Strings). See: https://appiumpro.com/editions/8

**Comment by KazuCocoa** [comment]
Appium uses [facebook/WebDriverAgent](https://github.com/facebook/WebDriverAgent) as a backend driver for over iOS 10 (`xcuitest` automation name)

https://github.com/facebook/WebDriverAgent/wiki/How-To-Achieve-The-Best-Lookup-Performance will help you to understand which strategy is heavy to lookup.

**Comment by titusfortner** [comment]
Yeah, essentially we'll want to create a `ClassChain` class that implements these methods: https://github.com/watir/watir/blob/master/lib/watir/locators/element/selector_builder/xpath.rb

It's going to be interesting figuring out how to properly deal with implementations that are browser/driver dependent (which Watir doesn't have to deal with)
